backend/models/Notes.spec.js

```diff
@@ -62,5 +61,49 @@ describe('Notes Model:', () => {
       expect(response.message).toBe("Note validation failed: author: Path `author` is required.");
     });
   });
+
+  describe('CRUD Tests:', () => {
+    /* Test Data */
+
+    it('saves a new note to the DB.', async () => {
+      let response;
+      try {
+        response = await Notes.create(testNote);
+        noteWithoutAuthor._id = response._id;
+      } catch(error) {
+        console.log('saves a new note to the DB--ERROR:',error);
+      }
+
+      expect(response).toMatchObject(noteWithoutAuthor);
+      // An error appears if we try to match the response with the test note directly
+      // The author fields might match, but the formats are different, with one a string
+      // and the other a full blown BSON object thing.
+    });
+
+    it('allows an existing note to be modified.', async () => {
+      const editedNote = { ...noteWithoutAuthor, text: 'thing object blown full fields match' };
+      let response;
+      try {
+        const idOfNoteToEdit = noteWithoutAuthor._id;
+        const configObj = { new: true }; // configure response to be the updated object
+        response = await Notes.findByIdAndUpdate(idOfNoteToEdit, editedNote, configObj);
+      } catch(error) {
+        console.log('saves a new note to the DB--ERROR:',error);
+      }
+      expect(response).toMatchObject(editedNote);
+    });
+
+    // it('removes an existing note from the DB.', async () => {
+    //   let removeResponse;
+    //   try {
+    //     const idOfNoteToRemove = noteWithoutAuthor._id;
+    //     removeReponse = await Notes.findByIdAndRemove(idOfNoteToRemove);
+    //   } catch (error) {
+    //     console.log('removes an existing note from the DB--ERROR:',error);
+    //   }
+    //   console.log('removes an existing note from the DB--removeResponse:',removeResponse);
+    //   expect(removeResponse).toMatchObject(noteWithoutAuthor);
+    // });
+  });
```

Comment:
Ronald,
It is awesome that you have commented your code. It will be easier to reference it later.